File: LeetCode Solutions/Problems/2395. Find Subarrays With Equal Sum.py
class Solution:
    def findSubarrays(self, nums: List[int]) -> bool:
        sums = set()
        for i in range(len(nums) - 1):
            getSum = nums[i] + nums[i + 1]
            if getSum in sums:
                return True
            sums.add(getSum)
        return False

File: LeetCode Solutions/Problems/2395. Find Subarrays With Equal Sum.cpp
class Solution {
public:
    bool findSubarrays(vector<int>& nums) {
        map<int, int> mp;
        for (int i=1; i<nums.size(); i++){
            int s = nums[i] + nums[i-1];
            if (mp.find(s) != mp.end()){
                return true;
            }
            mp[s]++;
        }
        return false;
    }
};